packages/ibm-products/src/components/ConditionBuilder/ConditionBuilderGroup/ConditionBuilderGroup.js

```diff
@@ -0,0 +1,170 @@
+                        ),
+                      });
+                    } else {
+                      onRemove();
+                    }
+                  }}
+                />
+              )}
+              {/* rendering each condition block */}
+              {!eachCondition.conditions && (
+                <>
+                  <ConditionBlock
+                    conjunction={
+                      conditionIndex > 0 ? state.groupOperator : undefined
+                    }
+                    aria={{
+                      level: aria.level + 1,
+                      posinset: conditionIndex + 1,
+                      setsize: state.conditions.length,
+                    }}
+                    isStatement={conditionIndex == 0}
+                    state={eachCondition}
+                    group={state}
+                    conditionIndex={conditionIndex}
+                    className={`${blockClass}__gap ${blockClass}__gap-bottom`}
+                    onChange={(updatedConditions) => {
+                      onChange({
+                        ...state,
+                        conditions: state.conditions.map((condition, cIndex) =>
+                          conditionIndex === cIndex
+                            ? updatedConditions
+                            : condition
+                        ),
+                      });
+                    }}
+                    onRemove={() => {
+                      if (state.conditions.length > 1) {
+                        onChange({
+                          ...state,
+                          conditions: state.conditions.filter(
+                            (c, cIndex) => conditionIndex !== cIndex
+                          ),
+                        });
+                      } else {
+                        onRemove();
+                      }
+                    }}
+                    onConnectorOperatorChange={(op) => {
+                      onChange({
+                        ...state,
+                        groupOperator: op,
+                      });
+                    }}
+                    onStatementChange={(updatedStatement) => {
+                      onChange({
+                        ...state,
+                        statement: updatedStatement,
+                      });
+                    }}
+                  ></ConditionBlock>
+                  {/* for last condition shows add button */}
+                  {isLastCondition(conditionIndex, conditionArr) && (
+                    <ConditionBuilderAdd
+                      onClick={() => {
+                        onChange({
+                          ...state,
+                          conditions: [
+                            ...state.conditions.slice(0, conditionIndex + 1),
+                            {
+                              property: undefined,
+                              operator: '',
+                              value: '',
+                              open: 'open',
```

Comment:
We cannot use boolean here. 
Condition builder will re-render with the updated data tree on user interactions. Here popoverState expects 3 values : open, close, ’’ to determine the default state of a popover

 ‘open’ : when  add a new condition
‘close’ : to close single select popover when an option is selected.Here we cannot not access  popover state or config directly as popover content is rendered as children. 
So, use effect is triggered because popoverState changes from ‘’ to ‘close’ when any option is selected
‘’ : default value.
Please see the [this link](https://deploy-preview-4951--carbon-for-ibm-products.netlify.app/?path=/story/ibm-products-components-conditionbuilder--condition-builder) of the deployment of complete changes . I have raised a https://github.com/carbon-design-system/ibm-products/pull/4951with all changes